src/Common/src/System/CharArrayHelpers.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System
+{
+    internal static class CharArrayHelpers
+    {
+        internal static bool EqualsOrdinal(string left, char[] right, int rightStartIndex, int rightLength)
+        {
+            Debug.Assert(left != null);
+            DebugAssertArrayInputs(right, rightStartIndex, rightLength);
+
+            if (left.Length != rightLength)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                if (left[i] != right[rightStartIndex + i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        internal static bool EqualsOrdinalAsciiIgnoreCase(string left, char[] right, int rightStartIndex, int rightLength)
+        {
+            Debug.Assert(left != null);
+            DebugAssertArrayInputs(right, rightStartIndex, rightLength);
+
+            if (left.Length != rightLength)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                uint charA = left[i];
+                uint charB = right[rightStartIndex + i];
+
+                // We're only interested in ASCII characters here.
+                if ((charA - 'a') <= ('z' - 'a')) charA -= 0x20;
+                if ((charB - 'a') <= ('z' - 'a')) charB -= 0x20;
+
+                if (charA != charB)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        internal static void Trim(char[] array, ref int startIndex, ref int length)
+        {
+            DebugAssertArrayInputs(array, startIndex, length);
+
+            int offset = 0;
+            while (offset < length && char.IsWhiteSpace(array[startIndex + offset]))
+            {
+                offset++;
+            }
+
+            int end = length - 1;
+            while (end >= offset && char.IsWhiteSpace(array[startIndex + end]))
+            {
+                end--;
+            }
+
+            startIndex += offset;
+            length = end - offset + 1;
+        }
```

Comment:
Doesn't need to be now, be we should look at moving this into Common along with code very close to it: https://github.com/dotnet/corefx/blob/master/src/Common/src/System/StringExtensions.cs